### ISSUE TITLE
Error: Protocol error (Target.createTarget): Target closed.

### ISSUE THREAD

**Issue by krunalshah68** [Kikobeats/aws-lambda-chrome#12]
When trying to run with AWS Lambda I'm getting this error.
`index.js` is same as required by AWS Lambda with `handler` function and requiring my other file in index.js.
Here is code from that file: 
```
// this one is same as example `get-browserless.js`
const getBrowserless = require('./get_browserless')

async function startTest() {
	try {
		const { text } = await getBrowserless();
		const url = 'https://google.com';
		const urlText = await text(url)
		console.log(urlText)
		return Promise.resolve();
	} catch (error) {
		console.log(error)
		return Promise.reject();
	}
}

const test = { startTest };
module.exports = test;
```

**Comment by Kikobeats** [comment]
What version of the library? Did you try with previous versions?

**Comment by krunalshah68** [comment]
Latest version 1.10.1.
Nope haven't tried with previous versions. Let me try that.

**Comment by Kikobeats** [comment]
The thing that I don't understand is that I'm using currently the library under my AWS Lambda on production

**Comment by krunalshah68** [comment]
umm can you post that sample code (including browserless and aws-chrome-lambda)? May be I can try that. :)

**Comment by Kikobeats** [comment]
First I have a singleton instance to ensure to use it globally

```js
'use strict'

const createBrowserless = require('browserless')

const { TMP_FOLDER, REQ_TIMEOUT } = require('../constant')

const KEY = '__browserless__'

const getChromePath = require('@browserless/aws-lambda-chrome')({
  path: TMP_FOLDER
})

const isLambda = !!process.env.LAMBDA_TASK_ROOT
const getExecutablePath = async () => (isLambda ? getChromePath() : undefined)

module.exports = async () =>
  global[KEY] ||
  (global[KEY] = createBrowserless({
    timeout: REQ_TIMEOUT,
    executablePath: await getExecutablePath()
  }))
```

Then calling it on action

```js
'use strict'

const getBrowserless = require('../../lib/get-browserless')


module.exports = async ({ url, ...opts }) => {
  const { screenshot: takeScreenshot } = await getBrowserless()
  const tmpStream = await takeScreenshot(url, getScreenshotOpts(opts))
}

```

**Comment by krunalshah68** [comment]
okay let me try that. Just one more question would it be possible to run 100 concurrent sessions of headless-chrome to load test a website on lambda?

**Comment by Kikobeats** [comment]
browserless supports [pool](https://browserless.js.org/#/?id=pooloptions) mode, but if you are using it at AWS Lambda it automatically scale the project based on requests demand, check you have it properly configured, that's should be enough

**Comment by krunalshah68** [comment]
Thanks again @Kikobeats
yep had complete setup with browserless pool and aws-lambda-chrome but ran to the issue that I've raised. The code that you're providing is almost the identical what I'm having but still having issues. Will try with very simple one now.

**Comment by Kikobeats** [comment]
If you enter at https://chat.microlink.io/ we can have a better conversation 🙂

I'm going to close the issue because looks a problem related with infrastructure, but please feel free to reply if you found something useful there!

**Comment by floydnoel** [comment]
Having the same issue, was there any resolution for @krunalshah68 ?

**Comment by krunalshah68** [comment]
@floydnoel unfortunately no. I've moved my project to EC2 :(
Let me know if you get it working!

**Comment by floydnoel** [comment]
I couldn't, so I ended up using https://github.com/sambaiz/puppeteer-lambda-starter-kit

**Comment by Kikobeats** [comment]
I investigated and this is a thing related with incognito and single process on AWS Lambda: https://github.com/alixaxel/chrome-aws-lambda/issues/8

Based on that and aim to provide a good library in terms of development, I released browserless v5.0.0 where incognito is disabled by default.

Also, I'm going to deprecate this library in favor of https://github.com/alixaxel/chrome-aws-lambda; It contents better default and more community, doesn't make too sense duplicate efforts in the same thing 🙂
